### PR TITLE
feat: add label filter

### PR DESCRIPTION
**Pull request by raphink**
Signed-off-by: Raphaël Pinson <raphael@isovalent.com>

### CODE REVIEW

**Comment by alvaroaleman** [comment]
Thanks! Unfortunatelly, this would only work with equaliity-based selectors, i.E. something like `kubectl get pods -l 'environment,environment notin (frontend)` wouldn't work as expected.

Can we maybe use k8s.io/apimachinery/pkg/labels.Parse here?

**Comment by raphink** [comment]
Ah sure, didn't think about other expressions.

**Comment by raphink** [comment]
Do you know how to retrieve the `Labels` interface from the `unstructured.Unstructured` structure?

**Comment by alvaroaleman** [comment]
I think `labels.Set(unstructured.GetLabels())` should work

**Comment by raphink** [comment]
There you go, this seems to work, just tested with `kubectl get po -l 'k8s-app in (cilium, redis-cart, hubble-ui)' -l pod-template-hash=5c58dcdc9c -A`

File: pkg/filter/filter.go [hunk 1]
```diff
@@ -50,3 +55,38 @@ func filterForFieldSelector(value []string) Filter {
...
+func filterForLabels(value []string) Filter {
+	var selectors []labels.Selector
+	return func(in *unstructured.UnstructuredList) (*unstructured.UnstructuredList, error) {
+		if len(value) == 0 {
+			return in, nil
+		}
+		for _, entry := range value {
+			selector, err := labels.Parse(entry)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse label selector: %s", entry)
```
Comment (alvaroaleman) [inline pkg/filter/filter.go:68]:
```suggestion
				return nil, fmt.Errorf("failed to parse label selector %s: %w", entry, err)
```

**Pull request merged**